### PR TITLE
Gs propagate

### PR DESCRIPTION
**Pull request by jaredthomas68**
# Propagate calculated LCOS and LCOA when doing optimizations and DOEs

The code was overwriting the LCOH value in the dictionaries being passed to the steel model. However, that overwrite was remaining until the next run of the code, at which point the LCOH would not update as calculated because if not -1 the LCOH is not overwritten. This PR corrects that bug by making a deep copy of the dictionary before preparing it to pass to the steel or ammonia models.

This PR also updates the locations of the resource files so there is a single location for storing them and that location is inside of `hopp/` so the resource files can be made accessible even on HPC system installations.

## Impacted areas of the software

GreenHEART `greenheart_simulation.py`

### CODE REVIEW

**Comment by jaredthomas68** [comment]
Tests are passing locally. I'll need to figure out why they are not passing on github

File: tests/greenheart/test_greensteel/test_greensteel.py [hunk 1]
```diff
@@ -7,6 +7,8 @@
     GreenHeartSimulationConfig,
 )
 
+from hopp import ROOT_DIR
```
Comment (bayc) [inline tests/greenheart/test_greensteel/test_greensteel.py:10]: Should this be used somewhere below?

File: tests/greenheart/test_openmdao.py [hunk 2]
```diff
@@ -21,6 +26,37 @@
 greenheart_config_onshore_filename = Path(__file__).absolute().parent / "input_files" / "plant" / "greenheart_config_onshore.yaml"
 turbine_config_filename = Path(__file__).absolute().parent / "input_files" / "turbines" / "osw_18MW.yaml"
 floris_input_filename_steel_ammonia = Path(__file__).absolute().parent / "input_files" / "floris" / "floris_input_osw_18MW.yaml"
+
+
+from ORBIT.core.library import initialize_library
+
+dirname = os.path.dirname(__file__)
+orbit_library_path = os.path.join(dirname, "input_files/")
+YamlIncludeConstructor.add_to_loader_class(
```
Comment (bayc) [inline tests/greenheart/test_openmdao.py:35]: Does the `load_yaml` function from `hopp.utilities` not work for these files? Not positive that we need to add a constructor for them.

File: tests/greenheart/test_openmdao.py [hunk 3]
```diff
@@ -379,3 +415,37 @@ def test_run_greenheart_optimize(subtests):
     
     with subtests.test("lcoh"):
         assert lcoh_final < lcoh_init
+
+# def test_greenheart_offshore_wind_wave_solar_battery():
```
Comment (bayc) [inline tests/greenheart/test_openmdao.py:419]: Do we want to keep these?

File: greenheart/simulation/technologies/steel/steel.py [hunk 4]
```diff
@@ -901,6 +901,13 @@ def run_steel_full_model(greenheart_config: dict, save_plots=False, show_plots=F
     # we'll just copy the config and modify it as needed
     config = copy.deepcopy(greenheart_config)
 
+    if config["steel"]["costs"]["lcoh"] != config["steel"]["finances"]["lcoh"]:
+        raise(ValueError(
+            "steel cost LCOH and steel finance LCOH are not equal. You must specify both values or neither. \
+                If neither is specified, LCOH will be calculated."
```
[the remark this answers is not in the document]
Comment (jaredthomas68) [reply greenheart/simulation/technologies/steel/steel.py:907]: I completely agree.

File: greenheart/tools/optimization/openmdao.py [hunk 5]
```diff
@@ -62,15 +62,14 @@ def setup(self):
             self.add_output("lcos", units="USD/t", val=0.0, desc="levelized cost of steel")
         if "ammonia" in self.options["config"].greenheart_config.keys():
             self.add_output("lcoa", units="USD/kg", val=0.0, desc="levelized cost of ammonia")
-        if "pv_capacity_kw" in self.options["design_variables"]:
-            if self.options['config'].design_scenario["pv_location"] == "offshore":
-                if self.options["config"].greenheart_config["opt_options"]["constraints"]["pv_to_platform_area_ratio"]["flag"]:
 
-                    self.add_output("pv_area", units="m*m", val=0.0, desc="offshore pv array area")
-                    self.add_output("platform_area", units="m*m", val=0.0, desc="offshore platform area")
+        if self.options["config"].greenheart_config["opt_options"]["constraints"]["pv_to_platform_area_ratio"]["flag"]:
```
[the remark this answers is not in the document]
Comment (jaredthomas68) [reply greenheart/tools/optimization/openmdao.py:66]: I was mimicking WISDEM. I'm open to discussion though. Leaving for now.

File: tests/greenheart/test_greensteel/test_greensteel.py (hunk 1 above)
Comment (jaredthomas68) [reply tests/greenheart/test_greensteel/test_greensteel.py:10]: Not anymore. Removed.

File: tests/greenheart/test_openmdao.py (hunk 2 above)
Comment (jaredthomas68) [reply tests/greenheart/test_openmdao.py:35]: I guess we don't need this. I've removed those lines.

File: tests/greenheart/test_openmdao.py (hunk 3 above)
Comment (jaredthomas68) [reply tests/greenheart/test_openmdao.py:419]: I may put it back in complete later, but not needed for now.

Comment (jaredthomas68) [reply tests/greenheart/test_openmdao.py:419]: Removed

**Review by bayc — APPROVED** [review]

**Pull request merged**